src/datasets.c

```diff
@@ -1607,13 +1604,11 @@ static int DatasetOpSerialized(Dataset *set, const char *string, DatasetOpFunc D
 
     switch (set->type) {
         case DATASET_TYPE_STRING: {
-            // coverity[alloc_strlen : FALSE]
-            uint8_t decoded[strlen(string)];
-            uint32_t consumed = 0, num_decoded = 0;
-            Base64Ecode code =
-                    DecodeBase64(decoded, (uint32_t)strlen(string), (const uint8_t *)string,
-                            (uint32_t)strlen(string), &consumed, &num_decoded, Base64ModeStrict);
-            if (code == BASE64_ECODE_ERR) {
+            uint32_t decoded_size = Base64DecodeBufferSize(strlen(string));
+            uint8_t decoded[decoded_size];
+            uint32_t num_decoded = Base64Decode(
+                    (const uint8_t *)string, strlen(string), 0, Base64ModeStrict, decoded);
+            if (num_decoded == 0 && strlen(string) > 0) {
```

Comment:
we could bail earlier if  strlen(string) == 0 ? (or *string == 0)

Reply:
indeed.